tests/run/type_inference.pyx

```diff
@@ -789,3 +789,16 @@ def test_bound_methods():
   default_arg = o.default_arg
   assert default_arg(2) == 12, default_arg(2)
   assert default_arg(2, 3) == 15, default_arg(2, 2)
+
+def test_builtin_max():
+    """
+    # builtin max is slightly complicated because it gets transformed to EvalWithTempExprNode
+    # GH4155
```

Comment:
While most problems _can_ be solved with an indirection, information largely benefits from direct access.
```suggestion
    # See https://github.com/cython/cython/issues/4155
```